<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QFRDRImageMaskByIntensity</class>
 <widget class="QDialog" name="QFRDRImageMaskByIntensity">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>954</width>
    <height>564</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mask from Image ...</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
     <item>
      <widget class="QFPlotter" name="pltMain" native="true"/>
     </item>
     <item>
      <layout class="QFormLayout" name="formLayout_2">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Data Range:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="labRange">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Threshold:</string>
         </property>
         <property name="buddy">
          <cstring>edtLevel</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="1">
          <widget class="QFDoubleEdit" name="edtLevel"/>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QFDoubleEdit" name="edtLevel_2"/>
         </item>
         <item row="1" column="1">
          <widget class="QFDoubleSlider" name="sldLevel">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QFDoubleSlider" name="sldLevel_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Mode:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QComboBox" name="cmbMaskingMode">
           <item>
            <property name="text">
             <string>mask in-range pixels</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>mask out-of-range pixels</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>mask smaller threshold</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>mask larger threshold</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="1">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="4" column="1">
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>&amp;update mask</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>mask m&amp;ode:</string>
         </property>
         <property name="buddy">
          <cstring>cmbMaskMode</cstring>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QComboBox" name="cmbMaskMode">
         <item>
          <property name="text">
           <string>replace</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>add</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>remove</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QFPlotter</class>
   <extends>QWidget</extends>
   <header>qfplotter.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QFDoubleEdit</class>
   <extends>QLineEdit</extends>
   <header>qfdoubleedit.h</header>
   <slots>
    <signal>valueChanged(double)</signal>
    <slot>setValue(double)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>QFDoubleSlider</class>
   <extends>QSlider</extends>
   <header>qfdoubleslider.h</header>
   <slots>
    <signal>dsliderMoved(double)</signal>
    <slot>setDValue(double)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>266</x>
     <y>554</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>554</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edtLevel</sender>
   <signal>editingFinished()</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>updateMask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>784</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edtLevel</sender>
   <signal>returnPressed()</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>updateMask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>784</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>408</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>updateMask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>942</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edtLevel</sender>
   <signal>textChanged(QString)</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>updateMask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>784</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>289</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbMaskingMode</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>updateMask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>941</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>582</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edtLevel_2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>updateMask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>829</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>579</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edtLevel</sender>
   <signal>valueChanged(double)</signal>
   <receiver>sldLevel</receiver>
   <slot>setDValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>702</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>700</x>
     <y>69</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldLevel</sender>
   <signal>dsliderMoved(double)</signal>
   <receiver>edtLevel</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>737</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>737</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edtLevel_2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>sldLevel_2</receiver>
   <slot>setDValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>938</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>935</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldLevel_2</sender>
   <signal>dsliderMoved(double)</signal>
   <receiver>edtLevel_2</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>914</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>916</x>
     <y>34</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldLevel</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>updateMask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>673</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>633</x>
     <y>69</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldLevel_2</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>QFRDRImageMaskByIntensity</receiver>
   <slot>updateMask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>821</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateMask()</slot>
  <slot>updateDualView()</slot>
 </slots>
</ui>
